lib/battle_boats/random_fleet_deployer.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "board"
+require_relative "coordinate"
+
+module BattleBoats
+  class RandomFleetDeployer
+    def initialize(board: BattleBoats::Board.new(state: :enemy))
+      @board = board
+    end
+
+    def deploy(fleet)
+      fleet.ships.each do |ship|
+        coin_flip = ["heads", "tails"].sample
+        if coin_flip == "heads"
+          until @board.place_ship_horizontally(coordinate: get_random_coordinate, ship: ship)
```

Comment:
It took me a while to understand exactly what was going on here. Does this `until` attempt to place ships on the `play_area` until it fails? What return values can we expect from this function. I think I might be confused, because this seems like a side effect, but we are relying on it's return value to make a decision... to do nothing.

Reply:
You're spot on. This is a result of me trying to be "clever."

Reply:
https://github.com/Thomascountz/battle_boats/pull/15